mava/environment_loop.py

```diff
@@ -575,24 +580,68 @@ def should_run_loop(eval_condtion: Tuple) -> bool:
                         "Running eval loop at executor step: "
                         + f"{self._last_evaluator_run_t}"
                     )
+
             return should_run_loop
 
         episode_count, step_count = 0, 0
 
-        # Currently, we only use intervals for eval loops.
-        environment_loop_schedule = (
-            self._executor._evaluator and self._executor._interval
-        )
-        if environment_loop_schedule:
-            eval_condition = check_count_condition(self._executor._interval)
+        # TODO (Ruan): Checking whether we are using Jax or TF here
+        # this should be removed once we deprecate TF.
+        if hasattr(self._executor, "store"):
+            environment_loop_schedule = self._executor._evaluator and (
+                self._executor.store._evaluation_interval is not None
+            )
+
+            if environment_loop_schedule:
+                eval_condition = check_count_condition(
+                    self._executor.store._evaluation_interval
+                )
+                evaluation_duration = self._executor.store._evaluation_duration
+        else:
+            environment_loop_schedule = (
+                self._executor._evaluator and self._executor._interval
+            )
+            if environment_loop_schedule:
+                eval_condition = check_count_condition(self._executor._interval)
 
         while not should_terminate(episode_count, step_count):
-            if (not environment_loop_schedule) or should_run_loop(eval_condition):
-                result = self.run_episode()
-                episode_count += 1
-                step_count += result["episode_length"]
-                # Log the given results.
-                self._logger.write(result)
+            if (not environment_loop_schedule) or (should_run_loop(eval_condition)):
+                # TODO (Ruan): Remove store check once TF is deprecated.
+                if environment_loop_schedule and hasattr(self._executor, "store"):
+                    # Get old battles won for true SMAC winrate
+                    if hasattr(self._environment, "battles_won"):
+                        old_battles_won = self._environment.battles_won
+
+                    # Get first result dictionary
+                    results = self.run_episode()
+                    episode_count += 1
+                    step_count += results["episode_length"]
+                    for _ in range(evaluation_duration - 1):
+                        # Add consecutive evaluation run data
+                        result = self.run_episode()
+                        episode_count += 1
+                        step_count += result["episode_length"]
+                        results = jax.tree_map(lambda x, y: x + y, results, result)
+
+                    # compute the mean over all evaluation runs
+                    results = jax.tree_map(lambda x: x / evaluation_duration, results)
+
+                    # Compute true win rate for SMAC
```

Comment:
I feel we should not have custom SMAC logic in our environment loop, but somehow include the logic in the SMAC wrapper.  We then call a generic environment function here. Not sure if this is possible?

Reply:
Done we have added a function called extra_ stats that will return the win rate from smac wrapper in the case of SMAC